### PR TITLE
Add contenttypes to lazy columns

### PR DESCRIPTION
**Pull request by MV-GH**
https://developer.android.com/jetpack/compose/lists#content-type

Adds content types to the lazy columns. From my limited testing, I did not notice any significant performance changes.

### CODE REVIEW

File: app/src/main/java/com/jerboa/ui/components/community/list/CommunityList.kt [hunk 1]
```diff
@@ -76,6 +76,7 @@ fun CommunityListings(
         items(
             communities,
             key = { it.community.id },
+            contentType = { "communitylink_${it.counts.users_active_month == 0}" },
```
Comment (dessalines) [inline app/src/main/java/com/jerboa/ui/components/community/list/CommunityList.kt:79]: just needs to be "communityLink"

**Review by dessalines — COMMENTED** [review]
According to the docs, this is just supposed to refer to the type of item, for lists which have different types. So it just needs to be "comment", "post", etc.

File: app/src/main/java/com/jerboa/ui/components/community/list/CommunityList.kt (hunk 1 above)
Comment (dessalines) [reply app/src/main/java/com/jerboa/ui/components/community/list/CommunityList.kt:79]: bump

File: app/src/main/java/com/jerboa/ui/components/comment/CommentNode.kt [hunk 2]
```diff
@@ -233,7 +233,7 @@ fun LazyListScope.commentNodeItem(
         .commentView.counts.child_count > 0 && !isFlat
 
     increaseLazyListIndexTracker()
-    item(key = commentId) {
+    item(key = commentId, contentType = { "contentNodeItemL${node.depth}" }) {
```
Comment (MV-GH) [inline app/src/main/java/com/jerboa/ui/components/comment/CommentNode.kt:236]: Eh it's just this one that worries me, I ll leave this one out for now and when I get some more time I ll try to get a microbench for this, bc the macrobench currently is not showing any significant perf changes. Same was for the scrolling with baselines/without but from personal expercience the baselines make a bigger impact than they are showing from the tests. So they aren't very reliable imo.

Comment (dessalines) [reply app/src/main/java/com/jerboa/ui/components/comment/CommentNode.kt]: Gotcha. Yep I'm guessing these aren't going to really matter until we have combined lists, like putting all the notification types (messages, comment replies, and mentions), in a single LazyList.

**Review by dessalines — APPROVED** [review]
Seems good, not sure why build is failing, I'll try a restart.

**Pull request merged**